python/caffe/test/test_net.py

```diff
@@ -79,3 +79,17 @@ def test_save_and_read(self):
             for i in range(len(self.net.params[name])):
                 self.assertEqual(abs(self.net.params[name][i].data
                     - net2.params[name][i].data).sum(), 0)
+
+    def test_save_hdf5(self):
+        f = tempfile.NamedTemporaryFile(mode='w+', delete=False)
+        f.close()
+        self.net.save_hdf5(f.name)
+        net_file = simple_net_file(self.num_output)
+        net2 = caffe.Net(net_file, caffe.TRAIN)
+        net2.load_hdf5(f.name)
```

Comment:
You're using camel case in the _caffe.cpp file (saveHDF5, loadHDF5), but snake case here. I think for consistency you should use save_hdf5, load_hdf5 in the _caffe.cpp file.